server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -160,17 +165,26 @@ public String executor() {
                 return ThreadPool.Names.GENERIC;
             }
         };
+
         try {
             // Will be replaced with ExtensionTokenProcessor and PrincipalIdentifierToken classes from feature/identity
             final String extensionTokenProcessor = "placeholder_token_processor";
             final String requestIssuerIdentity = "placeholder_request_issuer_identity";
 
+            this.allowList = List.of("Content-Type");
+            this.denyList = List.of("Authorization", "Proxy-Authorization");
+
+            Map<String, List<String>> filteredHeaders = headers.entrySet()
+                .stream()
+                .filter(e -> !denyList.contains(e.getKey()))
+                .filter(e -> allowList.contains("*") || allowlist.contains(e.getKey()))
```

Comment:
List.contains() is an O(N) search operation, the list needs to be scanned to find the element, change the underlying data type of allowList to be HashSet (Still immutable) so lookup time is O(1)

Reply:
> List.contains() is an O(N) search operation, the list needs to be scanned to find the element, change the underlying data type of allowList to be HashSet (Still immutable) so lookup time is O(1)

Looks like I can just replace `List.of` by `Set.of`

Reply:
Interesting... `Set.of()` is immutable and I am unable to find documentation on its underlying representation is from the java docs.

I asked Chat GPT, [transcript](https://gist.github.com/peternied/a00872db6a9b7ab8617077648134c342).  I didn't trust its answer so I double checked the implementation contains lookup check [[link]](https://hg.openjdk.org/jdk/jdk/file/ee1d592a9f53/src/java.base/share/classes/java/util/ImmutableCollections.java#l838) its a hashset - so O(1) runtime.

If you found this more easily I'd be curious where you found it. 🥂